main/test/ca/uwaterloo/flix/library/TestUnsafeBox.flix

```diff
@@ -33,6 +33,9 @@ namespace TestUnsafeBox {
     @test
     def testBox05(): Bool = box(None: Option[Int32]) == box(None: Option[Int32])
 
+    @test
+    def testBox06(): Bool = "${box(1 :: 2 :: 3 :: Nil)}" == "1 :: 2 :: 3 :: Nil"
```

Comment:
This can reproduce the typechecking bug in the Flix compiler we were talking about. Change this testcase to `def testBox06(): Bool = "${box(123)}" == "123"` and you get a bad cast exception at runtime.